packages/compiler/core/parser.ts

```diff
@@ -2181,9 +2181,13 @@ export function visitChildren<T>(node: Node, cb: NodeCb<T>): T | undefined {
         visitNode(cb, node.returnType)
       );
     case SyntaxKind.NamespaceStatement:
-      return visitEach(cb, node.decorators) || visitNode(cb, node.name) || isArray(node.statements)
-        ? visitEach(cb, node.statements as Statement[])
-        : visitNode(cb, node.statements);
+      return (
+        visitEach(cb, node.decorators) ||
+        visitNode(cb, node.name) ||
+        (isArray(node.statements)
+          ? visitEach(cb, node.statements as Statement[])
+          : visitNode(cb, node.statements))
```

Comment:
Classic mistake and it took me an embarrassing amount of time to spot this precedence problem. 😳

Reply:
The cast was a clue! Removed in follow-up commit.